packages/blitz/src/cli/utils/check-latest-version.ts

```diff
@@ -0,0 +1,185 @@
+import findUp from "find-up"
+import resolveFrom from "resolve-from"
+import {join, dirname} from "path"
+import fs from "fs"
+import {readVersions, resolveVersionType} from "./read-versions"
+import {getPkgManager} from "./helpers"
+import {log} from "../../logging"
+
+const BLITZ_ENDPOINT = `https://registry.npmjs.org/-/package/blitz/dist-tags?foo=bar`
+const BLITZ_AUTH_ENDPOINT = `https://registry.npmjs.org/-/package/@blitzjs/auth/dist-tags?foo=bar`
+const BLITZ_NEXT_ENDPOINT = `https://registry.npmjs.org/-/package/@blitzjs/next/dist-tags?foo=bar`
+const BLITZ_RPC_ENDPOINT = `https://registry.npmjs.org/-/package/@blitzjs/rpc/dist-tags?foo=bar`
+
+function getUpdateString(packageName: string, tag: string, isGlobal?: boolean) {
+  const pkgManager = getPkgManager()
+  switch (pkgManager) {
+    case "npm":
+      return `npm install ${isGlobal ? "-g" : ""} ${packageName}@${tag}`
+    case "yarn":
+      return `yarn ${isGlobal ? "global" : ""} add ${packageName}@${tag}`
+    case "pnpm":
+      return `pnpm install ${isGlobal ? "-g" : ""} ${packageName}@${tag}`
+  }
+}
+
+async function findNodeModulesRoot(src: string) {
+  let root: string
+  const blitzPkgLocation = dirname(
+    (await findUp("package.json", {
+      cwd: resolveFrom(src, "blitz"),
+    })) ?? "",
+  )
+
+  if (!blitzPkgLocation) {
+    throw new Error("Internal Blitz Error: unable to find 'blitz' package location")
+  }
+
+  if (blitzPkgLocation.includes(".pnpm")) {
+    root = join(blitzPkgLocation, "../../../../")
+  } else {
+    root = join(blitzPkgLocation, "../")
+  }
+
+  return root
+}
+
+export async function checkLatestVersion() {
+  const fetch = await import("node-fetch")
+  const boxen = await import("boxen")
+  const versions = readVersions()
+  const nodeModulesRoot = await findNodeModulesRoot(process.cwd())
+  const dotBlitzCacheExists = fs.existsSync(
+    join(nodeModulesRoot, ".blitz", "checkUpdateCache.json"),
+  )
+  let dotBlitzCache
+  let shouldRun
```

Comment:
You want to check if there is no cache

```suggestion
  let shouldRun = true
```